packages/lesswrong/components/comments/CommentsItem/ConvertToPostDialog.jsx

```diff
@@ -0,0 +1,114 @@
+import React, { PureComponent } from 'react';
+import { registerComponent, withDocument, withCreate, withUpdate } from 'meteor/vulcan:core';
+import Users from "meteor/vulcan:users";
+import Dialog from '@material-ui/core/Dialog';
+import DialogActions from '@material-ui/core/DialogActions';
+import DialogContent from '@material-ui/core/DialogContent';
+import DialogTitle from '@material-ui/core/DialogTitle';
+import Button from '@material-ui/core/Button';
+import Input from '@material-ui/core/Input';
+import Checkbox from '@material-ui/core/Checkbox';
+import { withStyles } from '@material-ui/core/styles';
+import withDialog from '../../common/withDialog'
+import withUser from '../../common/withUser'
+import { Posts } from '../../../lib/collections/posts';
+import { Comments } from '../../../lib/collections/comments';
+import { withNavigation } from '../../../lib/routeUtil'
+
+const styles = theme => ({
+  title: {
+    fontStyle: theme.typography.postStyle,
+    fontSize: "1.3rem"
+  }
+})
+
+class ConvertToPostDialog extends PureComponent {
```

Comment:
So, a thing I don't like about making this a dialog is that it forces you to make the move-comments decision before you've done your writing, and then you can't change it. But that's likely to be a multi-sitting writing task.